### ISSUE TITLE
SLA does not calculate correctly if you do not specify Monday AND Thursday (?)

### ISSUE THREAD

**Issue by Clam-** [osTicket/osTicket#6739]
### Prerequisites

* [ n/a ] Can you reproduce the problem in a fresh installation of the "develop" branch?
* [ N ] Do you have any errors in the PHP error log, or javascript console?
* [ Y ] Did you check the [osTicket forums](https://forum.osticket.com/)?
* [ Y ] Did you [perform a cursory search](https://github.com/osTicket/osTicket/issues) to see if your bug or enhancement is already reported?

### Description

When creating an SLA Schedule, if you do not specify Monday and Tuesday (for a weekly schedule), SLA timeframes are not calculated correctly.

### Steps to Reproduce

1. Create new SLA schedule
2. Add new Entry - enter Tuesday, 8am-5pm, starts on (some day in the past), Repeats Weekly, Day of the Week Tuesday
3. Click Diagnostics, and click Apply
Depending on the date you select on the diagnostics screen you'll either see no business days are counted or only one.

![SLA Schedule](https://github.com/user-attachments/assets/1ec8f304-231c-479a-85c3-4982e87d9cf7)

From what I can tell this will happen unless you have a Monday AND Thursday date set.

**Expected behavior:** SLA calculate normally.

**Actual behavior:** SLA calculates incorrectly and flags tickets overdue, sometimes instantly, sometimes on the "last day of the week".

### Versions
osTicket Version	v1.18.1 (0375576) —  Up to date
Web Server Software	LiteSpeed
MySQL Version	10.6.18
PHP Version	8.2.21

Hosted Cpanel, I do not know the OS. It's Linux based (3.10.0-962.3.2.lve1.5.81.el7.x86_64).
Chrome: Version 127.0.6533.89 (Official Build) (64-bit)

**Comment by d3sync** [comment]
Tracking the issue. I have some similar bug that after creating a custom schedule, as soon as I reply to an issue and keep the ticket open it's being marked as overdue regardless of the SLA.

**Comment by Clam-** [comment]
Yes, it was driving me crazy. I worked around it for now by creating 1 hour days on the other days since I had scope to be lenient.
It seems like creating 0 hour days should also work, since the diagnostic calculates time properly.
